subcommands/keys/rotate_root.go

```diff
@@ -0,0 +1,374 @@
+			if strings.TrimSpace(tk.KeyValue.Public) == pubkey {
+				pkbytes := creds[strings.Replace(k, ".pub", ".sec", 1)]
+				tk = client.TufKey{}
+				subcommands.DieNotNil(json.Unmarshal(pkbytes, &tk))
+				privPem, _ := pem.Decode([]byte(tk.KeyValue.Private))
+				if privPem == nil {
+					return nil, fmt.Errorf("Unable to parse private key: %s", string(creds[k]))
+				}
+				if privPem.Type != "RSA PRIVATE KEY" {
+					return nil, fmt.Errorf("Invalid private key???: %s", string(k))
+				}
+				pk, err := x509.ParsePKCS1PrivateKey(privPem.Bytes)
+				return pk, err
+			}
+		}
+	}
+	return nil, fmt.Errorf("Can not find private key for: %s", pubkey)
+}
+
+func findRoot(root client.TufRoot, creds OfflineCreds) (string, *rsa.PrivateKey, error) {
+	kid := root.Signed.Roles["root"].KeyIDs[0]
+	pub := root.Signed.Keys[kid].KeyValue.Public
+	key, err := findPrivKey(pub, creds)
+	return kid, key, err
+}
+
+func getOfflineCreds(credsFile string) OfflineCreds {
+	f, err := os.Open(credsFile)
+	subcommands.DieNotNil(err)
+	defer f.Close()
+
+	files := make(OfflineCreds)
+
+	gzf, err := gzip.NewReader(f)
+	subcommands.DieNotNil(err)
+	tr := tar.NewReader(gzf)
+
+	for {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			break // End of archive
+		}
+		subcommands.DieNotNil(err)
+
+		if hdr.Typeflag == tar.TypeDir {
+			continue
+		}
+
+		var b bytes.Buffer
+		_, err = io.Copy(&b, tr)
+		subcommands.DieNotNil(err)
+		files[hdr.Name] = b.Bytes()
+	}
+	return files
+}
+
+func syncProdRoot(factory string, curRoot client.TufRoot, creds OfflineCreds) error {
+	curProd, err := api.TufProdRootGet(factory)
+	if err != nil {
+		if httpE := client.AsHttpError(err); httpE != nil {
+			if httpE.Response.StatusCode == 404 {
+				// this is okay. it means we need to create the 1.rootjson
+				err = nil
+			}
+		}
+	}
+	if err != nil {
+		return err
+	}
+	prodVer := 0
+	if curProd != nil {
+		prodVer = curProd.Signed.Version
+	}
+	if prodVer == curRoot.Signed.Version {
+		return nil
+	}
+
+	for i := prodVer + 1; i <= curRoot.Signed.Version; i++ {
+		fmt.Println("= Populating production root version", i)
+		root, err := api.TufRootGetVer(factory, i)
+		if err != nil {
+			return err
+		}
+
+		// Bump the threshold
+		root.Signed.Roles["targets"].Threshold = 2
+
+		// Sign with the same keys used for the ci copy
+		var signers []TufSigner
+		for _, sig := range root.Signatures {
+			key, ok := root.Signed.Keys[sig.KeyId]
+			if !ok && i > 1 {
+				// Root key was rotated, this pub key is previous version
+				prev, err := api.TufRootGetVer(factory, i-1)
+				if err != nil {
+					return err
+				}
+				key = prev.Signed.Keys[sig.KeyId]
+			}
+			pkey, err := findPrivKey(key.KeyValue.Public, creds)
+			if err != nil {
+				return err
+			}
+			signers = append(signers, TufSigner{
+				Id:  sig.KeyId,
+				Key: pkey,
+			})
+		}
+		if err := signRoot(root, signers...); err != nil {
+			return err
+		}
+
+		bytes, err := json.MarshalIndent(root, "", "  ")
```

Comment:
Why indent here ?

Reply:
no real reason, when debugging it was helpful:

https://github.com/foundriesio/fioctl/pull/81/commits/1d68979748c697b652ea98cbc593d0104f661512